src/main/java/com/minecolonies/coremod/entity/ai/minimal/EntityAISleep.java

```diff
@@ -49,11 +64,51 @@ public boolean continueExecuting()
     {
         if (citizen.getDesiredActivity() == EntityCitizen.DesiredActivity.SLEEP)
         {
-            //TODO might search a bed?
+            final Colony colony = citizen.getColony();
+            if(colony == null)
+            {
+                return true;
+            }
+
+            if(usedBed.equals(BlockPos.ORIGIN))
```

Comment:
Please do not use BlockPos.ORIGIN to check if the bed has been set. What if the player builds it in a way that the Bed is at 0,0

Reply:
Is that possible, isn't that inside the bedrock?

Reply:
Theoretically not. It is not practical that it is there. But if for what ever reason one places a bed there we have a problem. So just use null makes it plain impossible to be mistaken.

Reply:
Or use an Optional.of if you desire.

Reply:
Question, Why is BlockPos.ORIGIN there anyway? what's it being used for?

Reply:
It is used to detect if the bed has been set. Which in my opinion is a bed way todo it. As it is theoritically possible to get a bed there, which would cause the citizen to never sleep.

Reply:
well what's another option for it?